app/controllers/cart_controller.rb

```diff
@@ -22,22 +22,31 @@ def remove_item
 
   def update_quantity
     item = Item.find(params[:item_id])
-    if params[:quantity] == "add"
-      if cart.contents["#{item.id}"] < item.inventory
-        cart.contents["#{item.id}"] += 1
-        redirect_to '/cart'
-      else
-        flash[:notice] = "No More Can Be Ordered At This Time"
-        redirect_to '/cart'
-      end
-    elsif params[:quantity] == "remove"
-      cart.contents["#{params[:item_id]}"] -= 1
-      if cart.contents["#{params[:item_id]}"] == 0
-        remove_item
-      else
-        redirect_to '/cart'
-      end
+    if params[:quantity] == 'add'
+      add_quantity(item)
+    elsif params[:quantity] == 'reduce'
+      reduce_quantity(item)
     end
   end
 
+  private
+
+  def add_quantity(item)
```

Comment:
I really like how you connected the add and reduce quantity methods to the car PORO